applications/minotari_app_grpc/src/authentication/basic_auth.rs

```diff
@@ -36,8 +41,17 @@ pub struct BasicAuthCredentials {
 }
 
 impl BasicAuthCredentials {
-    pub fn new(user_name: String, password: SafePassword) -> Self {
-        Self { user_name, password }
+    pub fn new(user_name: String, password: SafePassword) -> Result<Self, BasicAuthError> {
+        // Validate the username is well formed
+        if user_name.as_bytes().is_empty() || user_name.as_bytes().len() > MAX_USERNAME_LEN {
```

Comment:
Maybe a silly question, but is there a particular reason to forbid an empty username? It doesn't appear to be a requirement of RFC 2617, and security really shouldn't depend on username complexity.

Reply:
Ok, I will remove that restriction